pkg/webhook/register.go

```diff
@@ -200,6 +200,28 @@ func GetCommonWebhookConfigs() ([]Config, error) {
 	return whCfgs, nil
 }
 
+func getResourcesForImmutableValidation(allGVKs []schema.GroupVersionKind) []schema.GroupVersionKind {
+	// These IAM kinds have CEL validation for immutable fields and should be exempt from the webhook.
+	iamKindsWithCELValidation := map[string]bool{
+		"IAMAuditConfig":   true,
+		"IAMPartialPolicy": true,
+		"IAMPolicy":        true,
```

Comment:
Am I missing the immutable check on this? https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/v1.132.1/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_iampolicies.iam.cnrm.cloud.google.com.yaml#L137?

Either way it would be good to think about how we ensure that the field is immutable.

Reply:
I think we have some immutability scenario tests that @gemmahou worked on in the past